src/main/java/carpet/script/utils/ShapesRenderer.java

```diff
@@ -563,7 +563,7 @@ public void renderFaces(Tesselator tessellator, BufferBuilder bufferBuilder, dou
             else
                 RenderSystem.enableCull();
             
-            bufferBuilder.begin(shape.mode, DefaultVertexFormat.POSITION_COLOR);
+            bufferBuilder.begin(VertexFormat.Mode.values()[shape.mode], DefaultVertexFormat.POSITION_COLOR);
```

Comment:
Not reviewed yet, but mentioning that a call to Enum.values() makes a copy of the array. It should probably be cached somewhere instead.

Reply:
I would also not count on enum to stay correct forever, just list them up here in statics. Will do that after merge